### PR TITLE
bob: init at 0.5.3

### PR DESCRIPTION
**Pull request by zuzuleinen**
###### Description of changes

Add package for[ bob.build](https://bob.build/) a build system tool

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dit7ya — CHANGES_REQUESTED** [review]
Please remove the unintentional changes in the `all-packages.nix` file.

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (dit7ya) [inline pkgs/development/tools/build-managers/bob/default.nix:18]: Please add a comment on why the tests are failing/ disabled.

Comment (zuzuleinen) [reply pkgs/development/tools/build-managers/bob/default.nix:18]: The tests are failing for the following reason:

```shell
panic: failed to make git command: unable to find git executable in PATH; please install git before retrying
```

The code is using git at some point and is not found in PATH.  I wonder if I should add a comment about that or there is some way to make Git available.

Comment (dit7ya) [reply pkgs/development/tools/build-managers/bob/default.nix:18]: You can add `git` to `buildInputs`. If the tests requires network access then it won't work, and then just add a comment stating `tests require network access`. Otherwise you may also need `leaveDotGit` or `deepClone`. Search in nixpkgs with those keywords for various usage examples.

**Comment by dit7ya** [comment]
There is an output binary called `tui`, apart from `bob`. It does not seem to do anything when called. If it is not required for the main package you can omit it, for example using the `subPackages` variable.

**Comment by zuzuleinen** [comment]
> There is an output binary called `tui`, apart from `bob`. It does not seem to do anything when called. If it is not required for the main package you can omit it, for example using the `subPackages` variable.

Updated the `excludedPackages` and now it looks like tui binary is no longer created.

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (dit7ya) [inline pkgs/development/tools/build-managers/bob/default.nix:1]: `git` is no more required.

**Comment by dit7ya** [comment]
Result of `nixpkgs-review pr 189832` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bob</li>
  </ul>
</details>

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (sikmir) [inline pkgs/development/tools/build-managers/bob/default.nix:26]:
```suggestion
    maintainers = with maintainers; [ zuzuleinen ];
```
And add yourself to maintainer-list.

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (sikmir) [inline pkgs/development/tools/build-managers/bob/default.nix:14]:
```suggestion
  vendorHash = "sha256-R+zXGR5isoo76oc4lsFf9uCM0Kyi8dQiKEg4BUxtv+k=";
```

**Review by dit7ya — APPROVED** [review]
Looks good to me except the minor suggestions.

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (sikmir) [inline pkgs/development/tools/build-managers/bob/default.nix:11]:
```suggestion
    hash = "sha256-JG1fysCqqd/MwpNhKJwLr4cTGq4/88f9OMMapb+r3bc=";
```

**Comment by sikmir** [comment]
Please, squash commits.

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (sikmir) [reply pkgs/development/tools/build-managers/bob/default.nix:26]: Sure, https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

**Review by dit7ya — CHANGES_REQUESTED** [review]
The resulting binary returns a wrong value for `bob --version`. I think this is due to this - https://github.com/benchkram/bob/blob/6403a1de16304256725a06bba1522489ed1ccb27/.goreleaser.yml#L12

If `tag` is simply the version, you can use the `ldflags` attribute with `version` instead of `tag`. (also `-s` and `-w` should be used). And, following upstream, we should also set `CGO_ENABLED=0`.

Edit: Here's some documentation about the `ldflags`. Please write back if something is unclear. The docs are not optimal yet. 

https://nixos.org/manual/nixpkgs/unstable/#ssec-go-common-attributes

File: pkgs/development/tools/build-managers/bob/default.nix
Comment (dit7ya) [reply pkgs/development/tools/build-managers/bob/default.nix:26]: Here are some common starting points with contribution guidelines.

https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

**Comment by zuzuleinen** [comment]
Was not able to squash commits after merging master, will create another cleaner pr

**Comment by dit7ya** [comment]
Please add a link to this PR from the new one. Also, please try to avoid creating new PRs when possible - it makes reviewing a bit more difficult since context gets lost. Generally you don't need to merge master as chance of complex merge conflicts for new packages is pretty low. 

Without creating a new PR, you could locally clean up the branch and then force push to this PR's branch.

**Comment by zuzuleinen** [comment]
@dit7ya New PR: https://github.com/NixOS/nixpkgs/pull/189968

Sorry for the inconvenience

**Pull request closed without merging**
